packages/eslint-plugin/docs/rules/i18n-translator-comments.md

```diff
@@ -0,0 +1,38 @@
+# Enforce adding translator comments (i18n-translator-comments)
+
+If using [translation functions](https://github.com/WordPress/gutenberg/blob/master/packages/i18n/README.md#api) with placeholders in them,
+they need accompanying translator comments.
+
+## Rule details
+
+Examples of **incorrect** code for this rule:
+
+```js
+var color = '';
+sprintf( __( 'Color: %s' ), color );
+
+var address = '';
+sprintf(
+	__( 'Address: %s' ),
+	address
+);
+
+// translators: %s: Name
+var name = '';
+sprintf( __( 'Name: %s' ), name );
```

Comment:
These are interesting examples, because isn't it quite clear from the string itself what is expected from the placeholder? What more value are we providing through the translator comment than what is already available in the string by default?

I certainly understand there are cases which are clearly not so obvious by the string alone, in which case: Can we include those examples here instead, to better effectively demonstrate the problem we're trying to correct?

Thinking about what those cases are might also be a good thought exercise in considering if there are programmatic conditions we could derive from them (e.g. multiple placeholders, not preceded by `:`, etc).

Reply:
> These are interesting examples, because isn't it quite clear from the string itself what is expected from the placeholder?

I am pretty sure there are locales where you would translate `Name` (or something similar) differently depending on what name it is (person, block  title, website name, etc.)

This is also a reason why in core sometimes we add translator comments despite there not being any placeholder in a string, just to add extra clarification (e.g. `// translators: accessibility text`).

> Can we include those examples here instead, to better effectively demonstrate the problem we're trying to correct?

Sure, I can do that.

> Thinking about what those cases are might also be a good thought exercise in considering if there are programmatic conditions we could derive from them (e.g. multiple placeholders, not preceded by `:`, etc).

Are you thinking of something like checking whether the comment actually references all existing placeholders? Right now we should check whether it starts with `translators`

Reply:
> > Thinking about what those cases are might also be a good thought exercise in considering if there are programmatic conditions we could derive from them (e.g. multiple placeholders, not preceded by `:`, etc).
> 
> Are you thinking of something like checking whether the comment actually references all existing placeholders? Right now we should check whether it starts with `translators`

No, not this. The comment was in reference to an observation about the specific example.

The string is prefixed "Name:"
The translator comment we provided is "Name"

Is the string not clear enough on its own that the translator comment is actually improving anything?

I get that it's likely a contrived example here, but if we expect there to be real-world usage of this sort of string where the context of how a placeholder is used in a source string is pretty clear enough without the translator comment, if we could program this as exceptions of the rule, it would be worth exploring.

e.g.

```js
if ( /:\s+%.$/.test( string ) ) {
    // No need for translator comment.
    return;
}
```

But I guess that depends if the presence of a translator comment in this example would be harmful. Because otherwise, it's probably fine to err on the side of consistency.

Reply:
Got it.

I think that would make to many assumptions. As I said with my examples above, those translator comments are more often needed than one might think. Basically impossible to find exceptions programmatically.